lib/datastax/graph/structure.js

```diff
@@ -3,142 +3,45 @@
 const util = require("util");
 
 /**
- * @classdesc
- * Represents a graph Element.
- * @param id
- * @param label
- * @abstract
- * @memberOf module:datastax/graph
- * @constructor
+ * @deprecated
  */
 function Element(id, label) {
-  /**
-   * Gets the element id.
-   */
-  this.id = id;
-  /**
-   * Gets the element label.
-   * @type {String}
-   */
-  this.label = label;
+  throw new ReferenceError(`This class (Element) is not supported by our driver`);
 }
 
 /**
- * @classdesc
- * Represents a graph Vertex.
- * @param id
- * @param {String} label
- * @param {Object<string, Array>} properties
- * @extends {Element}
- * @memberOf module:datastax/graph
- * @constructor
+ * @deprecated
  */
 function Vertex(id, label, properties) {
-  Element.call(this, id, label);
-  /**
-   * Gets the vertex properties.
-   * @type {Object<string, Array>}
-   */
-  this.properties = properties;
+  throw new ReferenceError(`This class (Vertex) is not supported by our driver`);
 }
 
-util.inherits(Vertex, Element);
-
 /**
- * @classdesc
- * Represents a graph Edge.
- * @param id
- * @param outV
- * @param {outVLabel} outVLabel
- * @param {String} label
- * @param inV
- * @param {String} inVLabel
- * @param {Object<string, Property>} properties
- * @extends {Element}
- * @memberOf module:datastax/graph
- * @constructor
+ * @deprecated
  */
 function Edge(id, outV, outVLabel, label, inV, inVLabel, properties) {
-  Element.call(this, id, label);
-  /**
-   * Gets the id of outgoing vertex of the edge.
-   */
-  this.outV = outV;
-  /**
-   * Gets the label of the outgoing vertex.
-   */
-  this.outVLabel = outVLabel;
-  /**
-   * Gets the id of the incoming vertex of the edge.
-   */
-  this.inV = inV;
-
-  /**
-   * Gets the label of the incoming vertex.
-   */
-  this.inVLabel = inVLabel;
-  /**
-   * Gets the properties of the edge as an associative array.
-   * @type {Object}
-   */
-  this.properties = {};
-  (function adaptProperties(self) {
-    if (properties) {
-      const keys = Object.keys(properties);
-      for (let i = 0; i < keys.length; i++) {
-        const k = keys[i];
-        self.properties[k] = properties[k].value;
-      }
-    }
-  })(this);
+  throw new ReferenceError(`This class (Edge) is not supported by our driver`);
```

Comment:
One thought: If we'd like to change the error message at some point in the future, we will have to replace the string in all occurences. I suggest defining the message in one place and using it everywhere, so that changes in one place in code are necessary in such situation.

Reply:
The text could be parametrised by kind of the entity (`class`, `function`, etc.) and name of the entity (`Edge`).